var/spack/repos/builtin/packages/gmake/package.py

```diff
@@ -75,9 +75,14 @@ def configure_args(self):
 
     def install(self, spec, prefix):
         configure = Executable(join_path(self.stage.source_path, "configure"))
-        build_sh = Executable(join_path(self.stage.source_path, "build.sh"))
         with working_dir(self.build_directory, create=True):
             configure(f"--prefix={prefix}", *self.configure_args())
+            if spec.satisfies("@:4.2.1"):
+                # older configure creates build.sh in current directory
+                build_sh = Executable(join_path(".", "build.sh"))
+            else:
+                # newer configure creates build.sh in source directory
```

Comment:
Technically, this should probably be:
```suggestion
                # newer configure uses build.sh in source directory
```